exec.go

```diff
@@ -894,7 +894,16 @@ func runproc(win *Window, s string, dir string, newns bool, argaddr string, arg
 			shell = "rc"
 		}
 		rcarg = []string{shell, "-c", t}
+
+		env := os.Environ()
+		env = append(env, fmt.Sprintf("winid=%d", winid))
+		if filename != "" {
+			env = append(env, fmt.Sprintf("%%=%v", filename))
```

Comment:
This duplicates the environment setting code.  Could you extract the other occurences into a "setupenvvars" function so we have guaranteed consistency across these calls?

Reply:
@paul-lalonde Sure thing. Just pushed that up.